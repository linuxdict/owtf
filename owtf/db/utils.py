import math
from collections import namedtuple

from sqlalchemy.sql import and_, or_


def filter_none(kwargs):
    """
    Remove all `None` values froma  given dict. SQLAlchemy does not
    like to have values that are None passed to it.
    :param kwargs: Dict to filter
    :return: Dict without any 'None' values
    """
    n_kwargs = {}
    for k, v in kwargs.items():
        if v:
            n_kwargs[k] = v
    return n_kwargs


def session_query(model):
    """
    Returns a SQLAlchemy query object for the specified `model`.
    If `model` has a ``query`` attribute already, that object will be returned.
    Otherwise a query will be created and returned based on `session`.
    :param model: sqlalchemy model
    :return: query object for model
    """
    from owtf.db.session import get_scoped_session

    session = get_scoped_session()

    return model.query if hasattr(model, "query") else session.query(model)


def create_query(model, kwargs):
    """
    Returns a SQLAlchemy query object for specified `model`. Model
    filtered by the kwargs passed.
    :param model:
    :param kwargs:
    :return:
    """
    s = session_query(model)
    return s.filter_by(**kwargs)


def find_all(query, model, kwargs):
    """
    Returns a query object that ensures that all kwargs
    are present.
    :param query:
    :param model:
    :param kwargs:
    :return:
    """
    conditions = []
    kwargs = filter_none(kwargs)
    for attr, value in kwargs.items():
        if not isinstance(value, list):
            value = value.split(",")

        conditions.append(get_model_column(model, attr).in_(value))

    return query.filter(and_(*conditions))


def get_model_column(model, field):
    if field in getattr(model, "sensitive_fields", ()):
        raise AttrNotFound(field)
    column = model.__table__.columns._data.get(field, None)
    if column is None:
        raise AttrNotFound(field)

    return column


def find_any(query, model, kwargs):
    """
    Returns a query object that allows any kwarg
    to be present.
    :param query:
    :param model:
    :param kwargs:
    :return:
    """
    or_args = []
    for attr, value in kwargs.items():
        or_args.append(or_(get_model_column(model, attr) == value))
    exprs = or_(*or_args)
    return query.filter(exprs)


def filter(query, model, terms):
    """
    Helper that searched for 'like' strings in column values.
    :param query:
    :param model:
    :param terms:
    :return:
    """
    column = get_model_column(model, terms[0])
    return query.filter(column.ilike("%{}%".format(terms[1])))


def sort(query, model, field, direction):
    """
    Returns objects of the specified `model` in the field and direction
    given
    :param query:
    :param model:
    :param field:
    :param direction:
    """
    column = get_model_column(model, field)
    return query.order_by(column.desc() if direction == "desc" else column.asc())


def apply_pagination(query, page_number=None, page_size=None):
    """Apply pagination to a SQLAlchemy query object.
    :param page_number:
        Page to be returned (starts and defaults to 1).
    :param page_size:
        Maximum number of results to be returned in the page (defaults
        to the total results).
    :returns:
        A 2-tuple with the paginated SQLAlchemy query object and
        a pagination namedtuple.
        The pagination object contains information about the results
        and pages: ``page_size`` (defaults to ``total_results``),
        ``page_number`` (defaults to 1), ``num_pages`` and
        ``total_results``.
    Basic usage::
        query, pagination = apply_pagination(query, 1, 10)
        >>> len(query)
        10
        >>> pagination.page_size
        10
        >>> pagination.page_number
        1
        >>> pagination.num_pages
        3
        >>> pagination.total_results
        22
        >>> page_size, page_number, num_pages, total_results = pagination
    """
    total_results = query.count()
    query = _limit(query, page_size)

    # Page size defaults to total results
    if page_size is None or (page_size > total_results and total_results > 0):
        page_size = total_results

    query = _offset(query, page_number, page_size)

    # Page number defaults to 1
    if page_number is None:
        page_number = 1

    num_pages = _calculate_num_pages(page_number, page_size, total_results)

    Pagination = namedtuple("Pagination", ["page_number", "page_size", "num_pages", "total_results"])
    return query, Pagination(page_number, page_size, num_pages, total_results)


def _limit(query, page_size):
    if page_size is not None:
        if page_size < 0:
            raise Exception("Page size should not be negative: {}".format(page_size))

        query = query.limit(page_size)

    return query


def _offset(query, page_number, page_size):
    if page_number is not None:
        if page_number < 1:
            raise Exception("Page number should be positive: {}".format(page_number))

        query = query.offset((page_number - 1) * page_size)

    return query


def _calculate_num_pages(page_number, page_size, total_results):
    if page_size == 0:
        return 0

    return math.ceil(total_results / page_size)
