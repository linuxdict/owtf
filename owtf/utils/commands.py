"""
owtf.utils.commands
~~~~~~~~~~~~~~~~~~~

"""

import os


def get_command(argv):
    """Format command to remove directory and space-separated arguments.

    :params list argv: Arguments for the CLI.

    :return: Arguments without directory and space-separated arguments.
    :rtype: list

    """
    return " ".join(argv).replace(argv[0], os.path.basename(argv[0]))
