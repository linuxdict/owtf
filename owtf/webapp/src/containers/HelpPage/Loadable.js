/**
 * Asynchronously loads the component for HelpPage
 */
import Loadable from 'react-loadable';

import PageLoadingIndicator from 'components/PageLoadingIndicator';

export default Loadable({
  loader: () => import('./index'),
  loading: PageLoadingIndicator,
});
