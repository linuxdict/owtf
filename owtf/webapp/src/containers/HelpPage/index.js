/*
 * HelpPage
 */
import React from "react";
import { Pane, Text, Heading, Icon } from "evergreen-ui";

export default class Help extends React.Component {
  // Since state and props are static,
  // there's no need to re-render this component
  shouldComponentUpdate() {
    return false;
  }

  render() {
    const exploitationLinks = [
      { id: 1, text: "Hackvertor", link: "http://hackvertor.co.uk/public" },
      { id: 2, text: "ExploitDB", link: "http://www.exploit-db.com/" },
      { id: 3, text: "ExploitSearch", link: "http://www.exploitsearch.net/" },
      {
        id: 4,
        text: "Hackipedia",
        link: "http://www.hakipedia.com/index.php/Hakipedia"
      }
    ];

    const methodologyLinks = [
      {
        id: 1,
        text: "OWASP",
        link:
          "https://www.owasp.org/index.php/OWASP_Testing_Guide_v3_Table_of_Contents"
      },
      { id: 2, text: "Pentest Standard", link: "http://www.exploit-db.com/" },
      {
        id: 3,
        text: "OSSTMM",
        link: "http://www.isecom.org/research/osstmm.html"
      }
    ];

    const calculatorLinks = [
      {
        id: 1,
        text: "CVSS Advanced",
        link: "http://nvd.nist.gov/cvss.cfm?adv&calculator&version=2"
      },
      {
        id: 2,
        text: "CVSS Normal",
        link: "http://nvd.nist.gov/cvss.cfm?calculator&version=2"
      }
    ];

    const learnTestLinks = [
      {
        id: 1,
        text: "OWASP VWAD",
        link:
          "http://www.owasp.org/index.php?title=OWASP_Vulnerable_Web_Applications_Directory_Project"
      },
      {
        id: 2,
        text: "Securitythoughts",
        link:
          "http://securitythoughts.wordpress.com/2010/03/22/vulnerable-web-applications-for-learning/"
      },
      {
        id: 3,
        text: "Danielmiessler",
        link: "http://danielmiessler.com/projects/webappsec_testing_resources/"
      }
    ];

    const owtfHelpLinks = [
      [
        {
          id: 1,
          text: "Github Wiki",
          link: "https://github.com/owtf/owtf/wiki"
        },
        {
          id: 2,
          text: "Youtube channel",
          link: "http://www.youtube.com/user/owtfproject"
        },
        {
          id: 3,
          text: "Release notes",
          link: "http://blog.7-a.org/search/label/OWTF%20Release"
        }
      ],
      [
        {
          id: 1,
          text: "Github repository",
          link: "https://github.com/owtf/owtf"
        },
        {
          id: 2,
          text: "Issue tracker",
          link: "https://github.com/owtf/owtf/issues"
        }
      ],
      [
        {
          id: 1,
          text: "Mailing List",
          link: "https://lists.owasp.org/mailman/listinfo/owasp_owtf"
        },
        { id: 2, text: "Twitter", link: "https://twitter.com/owtfp" },
        { id: 3, text: "Medium Blog", link: "https://medium.com/@owtf" },
        { id: 4, text: "Author's blog", link: "http://blog.7-a.org" }
      ]
    ];

    return (
      <Pane
        clearfix
        display="flex"
        flexDirection="row"
        flexWrap="wrap"
        justifyContent="center"
        data-test="helpComponent"
      >
        <Pane
          elevation={1}
          width={350}
          margin={20}
          display="flex"
          flexDirection="column"
          data-test="helpBox"
        >
          <Heading height={40} background="#e1eaea">
            <Pane margin={10}>
              <Icon icon="search" /> Exploitation{" "}
            </Pane>
          </Heading>
          <Text margin={20}>
            <ul>
              {exploitationLinks.map(obj => (
                <li key={obj.id}>
                  <a href={obj.link} target="_blank" rel="noopener noreferrer">
                    {obj.text}
                  </a>
                </li>
              ))}
            </ul>
          </Text>
        </Pane>
        <Pane
          elevation={1}
          width={350}
          margin={20}
          display="flex"
          flexDirection="column"
          data-test="helpBox"
        >
          <Heading height={40} background="#e1eaea">
            <Pane margin={10}>
              <Icon icon="horizontal-bar-chart-desc" /> Methodology{" "}
            </Pane>
          </Heading>
          <Text margin={20}>
            <ul>
              {methodologyLinks.map(obj => (
                <li key={obj.id}>
                  <a href={obj.link} target="_blank" rel="noopener noreferrer">
                    {obj.text}
                  </a>
                </li>
              ))}
            </ul>
          </Text>
        </Pane>
        <Pane
          elevation={1}
          width={350}
          margin={20}
          display="flex"
          flexDirection="column"
          data-test="helpBox"
        >
          <Heading height={40} background="#e1eaea">
            <Pane margin={10}>
              <Icon icon="calculator" /> Calculators{" "}
            </Pane>
          </Heading>
          <Text margin={20}>
            <ul>
              {calculatorLinks.map(obj => (
                <li key={obj.id}>
                  <a href={obj.link} target="_blank" rel="noopener noreferrer">
                    {obj.text}
                  </a>
                </li>
              ))}
            </ul>
          </Text>
        </Pane>
        <Pane
          elevation={1}
          width={500}
          margin={20}
          display="flex"
          flexDirection="column"
          data-test="helpBox"
        >
          <Heading height={40} background="#e1eaea">
            <Pane margin={10}>
              <Icon icon="book" /> Test/Learn{" "}
            </Pane>
          </Heading>
          <Text margin={20}>
            <ul>
              {learnTestLinks.map(obj => (
                <li key={obj.id}>
                  <a href={obj.link} target="_blank" rel="noopener noreferrer">
                    {obj.text}
                  </a>
                </li>
              ))}
            </ul>
          </Text>
        </Pane>
        <Pane
          elevation={1}
          width={600}
          margin={20}
          display="flex"
          flexDirection="column"
          data-test="helpBox"
        >
          <Heading height={40} background="#e1eaea">
            <Pane margin={10}>
              <Icon icon="help" /> OWTF Help Links{" "}
            </Pane>
          </Heading>
          <Pane display="flex" flexDirection="row">
            <Pane display="flex" flexDirection="column">
              <Text margin={20}>
                <ul>
                  {owtfHelpLinks[0].map(obj => (
                    <li key={obj.id}>
                      <a
                        href={obj.link}
                        target="_blank"
                        rel="noopener noreferrer"
                      >
                        {obj.text}
                      </a>
                    </li>
                  ))}
                </ul>
              </Text>
            </Pane>
            <Pane display="flex" flexDirection="column">
              <Text margin={20}>
                <ul>
                  {owtfHelpLinks[1].map(obj => (
                    <li key={obj.id}>
                      <a
                        href={obj.link}
                        target="_blank"
                        rel="noopener noreferrer"
                      >
                        {obj.text}
                      </a>
                    </li>
                  ))}
                </ul>
              </Text>
            </Pane>
            <Pane display="flex" flexDirection="column">
              <Text margin={20}>
                <ul>
                  {owtfHelpLinks[2].map(obj => (
                    <li key={obj.id}>
                      <a
                        href={obj.link}
                        target="_blank"
                        rel="noopener noreferrer"
                      >
                        {obj.text}
                      </a>
                    </li>
                  ))}
                </ul>
              </Text>
            </Pane>
          </Pane>
        </Pane>
      </Pane>
    );
  }
}
