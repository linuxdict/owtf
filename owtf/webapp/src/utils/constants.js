/*
 * OWTF UI configuration
 *
 * This file will contain global configuration.
 */

export const API_BASE_URL = "/api/v1/";
export const API_BASE_ISSUE_URL = "https://github.com/owtf/owtf/issues/";
