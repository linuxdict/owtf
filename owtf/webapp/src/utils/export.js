/**
 * Funtion to import a directory.
 * :params r require.context object.
 * returns a object with key as filename and value as content of file.
 * If you need to import recursively, use true in second argument of request.context object.
 * Example usage importDirectory(require.context('<directory>', true, /\.(js|jsx|jpg)$/));
 */

export function importDirectory(r) {
  let elem = {};
  r.keys().map((item, index) => {
    elem[item.replace("./", "")] = r(item);
  });
  return elem;
}
