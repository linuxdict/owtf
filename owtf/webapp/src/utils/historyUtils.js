/**
 * historyUtils.js
 * This file returns a history object which can be used by both routers and sagas.
 */

import { createBrowserHistory } from "history";

const history = createBrowserHistory();

export default history;
