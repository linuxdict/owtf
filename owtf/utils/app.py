"""
owtf.utils.app
~~~~~~~~~~~~~~

"""
import tornado.web

try:
    from raven.contrib.tornado import AsyncSentryClient

    raven_installed = True
except ImportError:
    raven_installed = False

from owtf.settings import USE_SENTRY
from owtf.utils.error import get_sentry_client


class Application(tornado.web.Application):

    def __init__(self, *args, **kwargs):
        if raven_installed and USE_SENTRY:
            self.sentry_client = get_sentry_client()
        super(Application, self).__init__(*args, **kwargs)
