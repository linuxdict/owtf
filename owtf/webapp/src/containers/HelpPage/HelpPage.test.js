import React from 'react';
import { shallow } from 'enzyme';
import HelpPage from './index';
import '../../setupTests';
import toJson from 'enzyme-to-json';
import { findByTestAtrr } from '../../utils/testUtils';

const setUp = (props={}) => {
    const wrapper = shallow(<HelpPage {...props} />);
    return wrapper;
};

describe('HelpPage component', () => {

    let wrapper;
    beforeEach(() => {
      wrapper = setUp();
    });

    it('Should render without errors', () => {
        const component = findByTestAtrr(wrapper, "helpComponent");
        expect(component.length).toBe(1);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should successfully render sub-components', () => {
        const component = findByTestAtrr(wrapper, "helpBox");
        expect(component.length).toBe(5);
    })

});
