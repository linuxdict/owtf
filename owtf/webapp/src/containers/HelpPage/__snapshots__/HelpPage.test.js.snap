// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpPage component Should render without errors 1`] = `
<withTheme(Pane)
  clearfix={true}
  data-test="helpComponent"
  display="flex"
  flexDirection="row"
  flexWrap="wrap"
  justifyContent="center"
>
  <withTheme(Pane)
    data-test="helpBox"
    display="flex"
    elevation={1}
    flexDirection="column"
    margin={20}
    width={350}
  >
    <withTheme(Heading)
      background="#e1eaea"
      height={40}
    >
      <withTheme(Pane)
        margin={10}
      >
        <withTheme(Icon)
          icon="search"
        />
         Exploitation
         
      </withTheme(Pane)>
    </withTheme(Heading)>
    <withTheme(Text)
      margin={20}
    >
      <ul>
        <li
          key="1"
        >
          <a
            href="http://hackvertor.co.uk/public"
            rel="noopener noreferrer"
            target="_blank"
          >
            Hackvertor
          </a>
        </li>
        <li
          key="2"
        >
          <a
            href="http://www.exploit-db.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            ExploitDB
          </a>
        </li>
        <li
          key="3"
        >
          <a
            href="http://www.exploitsearch.net/"
            rel="noopener noreferrer"
            target="_blank"
          >
            ExploitSearch
          </a>
        </li>
        <li
          key="4"
        >
          <a
            href="http://www.hakipedia.com/index.php/Hakipedia"
            rel="noopener noreferrer"
            target="_blank"
          >
            Hackipedia
          </a>
        </li>
      </ul>
    </withTheme(Text)>
  </withTheme(Pane)>
  <withTheme(Pane)
    data-test="helpBox"
    display="flex"
    elevation={1}
    flexDirection="column"
    margin={20}
    width={350}
  >
    <withTheme(Heading)
      background="#e1eaea"
      height={40}
    >
      <withTheme(Pane)
        margin={10}
      >
        <withTheme(Icon)
          icon="horizontal-bar-chart-desc"
        />
         Methodology
         
      </withTheme(Pane)>
    </withTheme(Heading)>
    <withTheme(Text)
      margin={20}
    >
      <ul>
        <li
          key="1"
        >
          <a
            href="https://www.owasp.org/index.php/OWASP_Testing_Guide_v3_Table_of_Contents"
            rel="noopener noreferrer"
            target="_blank"
          >
            OWASP
          </a>
        </li>
        <li
          key="2"
        >
          <a
            href="http://www.exploit-db.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Pentest Standard
          </a>
        </li>
        <li
          key="3"
        >
          <a
            href="http://www.isecom.org/research/osstmm.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            OSSTMM
          </a>
        </li>
      </ul>
    </withTheme(Text)>
  </withTheme(Pane)>
  <withTheme(Pane)
    data-test="helpBox"
    display="flex"
    elevation={1}
    flexDirection="column"
    margin={20}
    width={350}
  >
    <withTheme(Heading)
      background="#e1eaea"
      height={40}
    >
      <withTheme(Pane)
        margin={10}
      >
        <withTheme(Icon)
          icon="calculator"
        />
         Calculators
         
      </withTheme(Pane)>
    </withTheme(Heading)>
    <withTheme(Text)
      margin={20}
    >
      <ul>
        <li
          key="1"
        >
          <a
            href="http://nvd.nist.gov/cvss.cfm?adv&calculator&version=2"
            rel="noopener noreferrer"
            target="_blank"
          >
            CVSS Advanced
          </a>
        </li>
        <li
          key="2"
        >
          <a
            href="http://nvd.nist.gov/cvss.cfm?calculator&version=2"
            rel="noopener noreferrer"
            target="_blank"
          >
            CVSS Normal
          </a>
        </li>
      </ul>
    </withTheme(Text)>
  </withTheme(Pane)>
  <withTheme(Pane)
    data-test="helpBox"
    display="flex"
    elevation={1}
    flexDirection="column"
    margin={20}
    width={500}
  >
    <withTheme(Heading)
      background="#e1eaea"
      height={40}
    >
      <withTheme(Pane)
        margin={10}
      >
        <withTheme(Icon)
          icon="book"
        />
         Test/Learn
         
      </withTheme(Pane)>
    </withTheme(Heading)>
    <withTheme(Text)
      margin={20}
    >
      <ul>
        <li
          key="1"
        >
          <a
            href="http://www.owasp.org/index.php?title=OWASP_Vulnerable_Web_Applications_Directory_Project"
            rel="noopener noreferrer"
            target="_blank"
          >
            OWASP VWAD
          </a>
        </li>
        <li
          key="2"
        >
          <a
            href="http://securitythoughts.wordpress.com/2010/03/22/vulnerable-web-applications-for-learning/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Securitythoughts
          </a>
        </li>
        <li
          key="3"
        >
          <a
            href="http://danielmiessler.com/projects/webappsec_testing_resources/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Danielmiessler
          </a>
        </li>
      </ul>
    </withTheme(Text)>
  </withTheme(Pane)>
  <withTheme(Pane)
    data-test="helpBox"
    display="flex"
    elevation={1}
    flexDirection="column"
    margin={20}
    width={600}
  >
    <withTheme(Heading)
      background="#e1eaea"
      height={40}
    >
      <withTheme(Pane)
        margin={10}
      >
        <withTheme(Icon)
          icon="help"
        />
         OWTF Help Links
         
      </withTheme(Pane)>
    </withTheme(Heading)>
    <withTheme(Pane)
      display="flex"
      flexDirection="row"
    >
      <withTheme(Pane)
        display="flex"
        flexDirection="column"
      >
        <withTheme(Text)
          margin={20}
        >
          <ul>
            <li
              key="1"
            >
              <a
                href="https://github.com/owtf/owtf/wiki"
                rel="noopener noreferrer"
                target="_blank"
              >
                Github Wiki
              </a>
            </li>
            <li
              key="2"
            >
              <a
                href="http://www.youtube.com/user/owtfproject"
                rel="noopener noreferrer"
                target="_blank"
              >
                Youtube channel
              </a>
            </li>
            <li
              key="3"
            >
              <a
                href="http://blog.7-a.org/search/label/OWTF%20Release"
                rel="noopener noreferrer"
                target="_blank"
              >
                Release notes
              </a>
            </li>
          </ul>
        </withTheme(Text)>
      </withTheme(Pane)>
      <withTheme(Pane)
        display="flex"
        flexDirection="column"
      >
        <withTheme(Text)
          margin={20}
        >
          <ul>
            <li
              key="1"
            >
              <a
                href="https://github.com/owtf/owtf"
                rel="noopener noreferrer"
                target="_blank"
              >
                Github repository
              </a>
            </li>
            <li
              key="2"
            >
              <a
                href="https://github.com/owtf/owtf/issues"
                rel="noopener noreferrer"
                target="_blank"
              >
                Issue tracker
              </a>
            </li>
          </ul>
        </withTheme(Text)>
      </withTheme(Pane)>
      <withTheme(Pane)
        display="flex"
        flexDirection="column"
      >
        <withTheme(Text)
          margin={20}
        >
          <ul>
            <li
              key="1"
            >
              <a
                href="https://lists.owasp.org/mailman/listinfo/owasp_owtf"
                rel="noopener noreferrer"
                target="_blank"
              >
                Mailing List
              </a>
            </li>
            <li
              key="2"
            >
              <a
                href="https://twitter.com/owtfp"
                rel="noopener noreferrer"
                target="_blank"
              >
                Twitter
              </a>
            </li>
            <li
              key="3"
            >
              <a
                href="https://medium.com/@owtf"
                rel="noopener noreferrer"
                target="_blank"
              >
                Medium Blog
              </a>
            </li>
            <li
              key="4"
            >
              <a
                href="http://blog.7-a.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                Author's blog
              </a>
            </li>
          </ul>
        </withTheme(Text)>
      </withTheme(Pane)>
    </withTheme(Pane)>
  </withTheme(Pane)>
</withTheme(Pane)>
`;
